### PR TITLE
feat: promote config_connector_config to ga

### PR DESCRIPTION
**Pull request by Tolsee**
- promote configConnectorConfig to ga
- Ref: https://github.com/hashicorp/terraform-provider-google/releases/tag/v4.49.0

### CODE REVIEW

**Comment by ericyz** [comment]
LGTM and will approve after the integration test passed

**Comment by Tolsee** [comment]
@ericyz Updated as per your comment. Thanks!

**Comment by Tolsee** [comment]
@bharathkkb Can you please let me know what is the failure here?

**Comment by ericyz** [comment]
golden.go:158: 
        	Error Trace:	/workspace/test/integration/simple_regional/golden.go:158
        	            				/workspace/test/integration/simple_regional/simple_regional_test.go:57
        	            				/workspace/test/integration/simple_regional/terraform.go:412
        	            				/workspace/test/integration/simple_regional/terraform.go:432
        	            				/workspace/test/integration/simple_regional/stages.go:31
        	            				/workspace/test/integration/simple_regional/terraform.go:432
        	Error:      	Not equal: 
        	            	expected: "{\n    \"dnsCacheConfig\": {},\n    \"gcePersistentDiskCsiDriverConfig\": {\n      \"enabled\": true\n    },\n    \"gcpFilestoreCsiDriverConfig\": {},\n    \"gkeBackupAgentConfig\": {},\n    \"horizontalPodAutoscaling\": {},\n    \"httpLoadBalancing\": {},\n    \"kubernetesDashboard\": {\n      \"disabled\": true\n    },\n    \"networkPolicyConfig\": {\n      \"disabled\": true\n    }\n  }"
        	            	actual  : "{\n    \"configConnectorConfig\": {},\n    \"dnsCacheConfig\": {},\n    \"gcePersistentDiskCsiDriverConfig\": {\n      \"enabled\": true\n    },\n    \"gcpFilestoreCsiDriverConfig\": {},\n    \"gkeBackupAgentConfig\": {},\n    \"horizontalPodAutoscaling\": {},\n    \"httpLoadBalancing\": {},\n    \"kubernetesDashboard\": {\n      \"disabled\": true\n    },\n    \"networkPolicyConfig\": {\n      \"disabled\": true\n    }\n  }"
        	            	
        	            	Diff:
        	            	--- Expected
        	            	+++ Actual
        	            	@@ -1,2 +1,3 @@
        	            	 {
        	            	+    "configConnectorConfig": {},
        	            	     "dnsCacheConfig": {},
        	Test:       	TestSimpleRegional
        	Messages:   	expected addonsConfig to match fixture {
        	            	    "dnsCacheConfig": {},
        	            	    "gcePersistentDiskCsiDriverConfig": {
        	            	      "enabled": true
        	            	    },
        	            	    "gcpFilestoreCsiDriverConfig": {},
        	            	    "gkeBackupAgentConfig": {},
        	            	    "horizontalPodAutoscaling": {},
        	            	    "httpLoadBalancing": {},
        	            	    "kubernetesDashboard": {
        	            	      "disabled": true
        	            	    },
        	            	    "networkPolicyConfig": {
        	            	      "disabled": true
        	            	    }
        	            	  }

**Comment by Tolsee** [comment]
@ericyz Can you help running it again. I did update the test https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/pull/1559/commits/d57285e60c5201e9278e1b81c9ad0ddc443f2beb

**Review by ericyz — APPROVED** [review]
LGTM

**Comment by ericyz** [comment]
@apeabody - I might need your help to peer review and merge at your convenience. Thanks

File: autogen/main/cluster.tf.tmpl
Comment (apeabody) [inline autogen/main/cluster.tf.tmpl:277]: Just a thought @ericyz, it might be possible to use a dynamic block to avoid breaking change for non-beta clusters.  Something like this: 

```
    dynamic "config_connector_config" {
      for_each = var.config_connector ? [{ enabled = true }] : []

      content {
        enabled = true
      }
    }
 ```
 
 LMKWYT - Would require additional changes for the beta clusters and tests, so I'm comfortable with this path as well.

**Review by apeabody — COMMENTED** [review]
Thanks for the contribution @Tolsee!

File: autogen/main/cluster.tf.tmpl
Comment (ericyz) [reply autogen/main/cluster.tf.tmpl:287]: Good suggestion. It's a good approach without waiting for next release

**Pull request merged**
